joomla/joomla-cms
# [4.0] Separate sources from the distributable assets

Pull Request for Issue # .

### Summary of Changes
This PR moves the static assets sources to the build directory.
There are 2 folders:
- media is holding all the legacy (eg ES5 js and css that don't have a relevant scss file)
- media_src is holding all the ES6 scripts and SCSS files
- the source for the templates remains in their respected template folders

There is no media folder anymore in the source and thus any change in this directory will not be flagged in git.

There is one more step now in the build process: creating the SPA for media manager

### Testing Instructions
Before fetching this branch make a copy of the folder media somewhere.
after fetch/pull you should run npm i and compare the 2 directories.
(the template assets weren't moved so nothing to check there!)


### Expected result
All assets regenerated with `npm i`, stop tracking compiled/minified files

### Documentation Changes Required
All the sources for the static assets are held in the `/build` folder.

PS. the `--compile-js` argument **will not** minify css and js files in the legacy folder for this you'll need to use the `--update` argument
